doc/neps/nep-0035-array-creation-dispatch-with-array-function.rst

```diff
@@ -8,16 +8,154 @@ NEP 35 — Array Creation Dispatching With __array_function__
 :Status: Draft
 :Type: Standards Track
 :Created: 2019-10-15
-:Updated: 2020-08-06
+:Updated: 2020-08-17
 :Resolution:
 
 Abstract
 --------
 
 We propose the introduction of a new keyword argument ``like=`` to all array
-creation functions to permit dispatching of such functions by the
-``__array_function__`` protocol, addressing one of the protocol shortcomings,
-as described by NEP-18 [1]_.
+creation functions, this argument permits the creation of an array based on
+a non-NumPy reference array passed via that argument, resulting in an array
+defined by the downstream library implementing that type, which also implements
+the ``__array_function__`` protocol. With this we address one of that
+protocol's shortcomings, as described by NEP 18 [1]_.
+
+Motivation and Scope
+--------------------
+
+Many are the libraries implementing the NumPy API, such as Dask for graph
+computing, CuPy for GPGPU computing, xarray for N-D labeled arrays, etc. All
+the libraries mentioned have yet another thing in common: they have also adopted
+the ``__array_function__`` protocol. The protocol defines a mechanism allowing a
+user to directly use the NumPy API as a dispatcher based on the input array
+type. In essence, dispatching means users are able to pass a downstream array,
+such as a Dask array, directly to one of NumPy's compute functions, and NumPy
+will be able to automatically recognize that and send the work back to Dask's
+implementation of that function, which will define the return value. For
+example:
+
+.. code:: python
+
+    x = dask.array.arange(5)    # Creates dask.array
+    np.sum(a)                   # Returns dask.array
```

Comment:
This is probably `np.sum(x)` right? I think a better candidate function would be np.diff or np.exp since the result would still be obviously an array. Here a sum might reduce down to a scalar.

Reply:
Assuming dask has `diff` or `exp` obviously.

Reply:
Agreed, it feels more natural to use something that returns an array by NumPy as well. It's not relevant for the discussion here or the NEP itself, but just wanted to note that Dask is doing lazy computation, so `np.sum(x)` would still return a Dask array, you would require `np.sum(x).compute()` to get the value of the sum.